src/databricks/labs/ucx/workspace_access/tacl.py

```diff
@@ -85,6 +94,9 @@ def get_apply_task(self, item: Permissions, migration_state: GroupMigrationState
             # this is a grant for user, service principal, or irrelevant group
             return None
         target_grant = dataclasses.replace(grant, principal=target_principal)
-        sql = target_grant.hive_grant_sql()
+        if grant.action_type.upper() == "OWN":
+            sql = target_grant.hive_chown_sql()
+        else:
+            sql = target_grant.hive_grant_sql()
```

Comment:
If you change only this method, then the amount of changed lines would probably reduce to 3 or 4.